File: plugins/tauri-plugin-sqlite/ios/Sources/ios/Args.swift
//
//  File.swift
//  
//
//  Created by SY L on 8/6/24.
//

import Foundation

struct PingReq: Decodable {
    let value: String?
}

struct PingRes: Encodable {
    let value: String?
}

struct GetDbUserVersionRes: Encodable {
    let version: Int32?
}

struct GetAllTodoRes: Encodable {
    let todos: [Todo]
}

struct InsertTodoReq: Decodable {
    let todo: String
}

struct InsertTodoRes: Encodable {
    let rowid: Int64
}

File: plugins/tauri-plugin-sqlite/src/args.rs
use serde::{Deserialize, Serialize};

use crate::Todo;

#[derive(Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct PingRequest {
  pub value: Option<String>,
}

#[derive(Debug, Clone, Default, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct PingResponse {
  pub value: Option<String>,
}


#[derive(Debug, Clone, Default, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetDbUserVersionRes {
    pub version: Option<u32>
}


#[derive(Debug, Clone, Default, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetAllTodoRes {
    todos: Vec<Todo>
}

#[derive(Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct InsertTodoReq {
    todo: String
}

#[derive(Debug, Clone, Default, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct InsertTodoRes {
    rowid: u32,
}
